40 tests fail on ARM32 (armv7hf): AssertionError: Got unexpected return code -4

**Describe the bug**
I try to build `0.8.0-rc1` for ALT Linux and found that 40 tests failed on armv7hf arch.
```
[00:00:48] + timeout 222 /usr/bin/ninja -j32 -C build run_tests
[00:00:48] ninja: Entering directory `build'
[00:00:49] [0/1] cd /usr/src/RPM/BUILD/liboqs-0.8.0.rc1 && /usr/bin/cmake -E env OQS_BUILD_DIR=/usr/src/RPM/BUILD/liboqs-0.8.0.rc1/build python3 -m pytest --verbose --numprocesses=auto --ignore=scripts/copy_from_upstream/repos
[00:00:55] ============================= test session starts ==============================
[00:00:55] platform linux -- Python 3.10.8, pytest-7.3.1, pluggy-1.0.0 -- /usr/bin/python3
[00:00:55] cachedir: .pytest_cache
[00:00:55] rootdir: /usr/src/RPM/BUILD/liboqs-0.8.0.rc1
[00:00:55] plugins: xdist-3.3.1
[00:00:55] created: 32/32 workers
[00:00:55] 32 workers [402 items]
...
[00:01:30] =================================== FAILURES ===================================
[00:01:30] ______________________ test_kem[Classic-McEliece-348864] _______________________
[00:01:30] [gw26] linux -- Python 3.10.8 /usr/bin/python3
[00:01:30] 
[00:01:30] kem_name = 'Classic-McEliece-348864'
[00:01:30] 
[00:01:30]     @helpers.filtered_test
[00:01:30]     @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
[00:01:30]     def test_kem(kem_name):
[00:01:30]         kats = helpers.get_kats("kem")
[00:01:30]         if not(helpers.is_kem_enabled_by_name(kem_name)): pytest.skip('Not enabled')
[00:01:30] >       helpers.run_subprocess( [helpers.path_to_executable('speed_kem'), kem_name, "-f"] )
[00:01:30] 
[00:01:30] tests/test_speed.py:14: 
[00:01:30] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[00:01:30] 
[00:01:30] command = ['/usr/src/RPM/BUILD/liboqs-0.8.0.rc1/build/tests/speed_kem', 'Classic-McEliece-348864', '-f']
[00:01:30] working_dir = '.'
[00:01:30] env = {'G_BROKEN_FILENAMES': '1', 'HISTFILESIZE': '9999', 'HISTSIZE': '999', 'HOME': '/usr/src', ...}
[00:01:30] expected_returncode = 0, input = None, ignore_returncode = False
[00:01:30] 
[00:01:30]     def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
[00:01:30]         """
[00:01:30]         Helper function to run a shell command and report success/failure
[00:01:30]         depending on the exit status of the shell command.
[00:01:30]         """
[00:01:30]         env_ = os.environ.copy()
[00:01:30]         if env is not None:
[00:01:30]             env_.update(env)
[00:01:30]         env = env_
[00:01:30]     
[00:01:30]         # Note we need to capture stdout/stderr from the subprocess,
[00:01:30]         # then print it, which pytest will then capture and
[00:01:30]         # buffer appropriately
[00:01:30]         print(working_dir + " > " + " ".join(command))
[00:01:30]     
[00:01:30]         result = subprocess.run(
[00:01:30]                 command,
[00:01:30]                 input=input,
[00:01:30]                 stdout=subprocess.PIPE,
[00:01:30]                 stderr=subprocess.STDOUT,
[00:01:30]                 cwd=working_dir,
[00:01:30]                 env=env,
[00:01:30]             )
[00:01:30]     
[00:01:30]         if not(ignore_returncode) and (result.returncode != expected_returncode):
[00:01:30]             print(result.stdout.decode('utf-8'))
[00:01:30] >           assert False, "Got unexpected return code {}".format(result.returncode)
[00:01:30] E           AssertionError: Got unexpected return code -4
[00:01:30] 
[00:01:30] tests/helpers.py:42: AssertionError
[00:01:30] ----------------------------- Captured stdout call -----------------------------
[00:01:30] . > /usr/src/RPM/BUILD/liboqs-0.8.0.rc1/build/tests/speed_kem Classic-McEliece-348864 -f
[00:01:30] 
...
...
[00:01:30] =========================== short test summary info ============================
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-348864] - AssertionErro...
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-348864f] - AssertionErr...
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-460896f] - AssertionErr...
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-6960119] - AssertionErr...
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-460896] - AssertionErro...
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-6688128f] - AssertionEr...
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-6960119f] - AssertionEr...
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-8192128] - AssertionErr...
[00:01:30] FAILED tests/test_speed.py::test_kem[HQC-192] - AssertionError: Got unexpecte...
[00:01:30] FAILED tests/test_speed.py::test_kem[HQC-128] - AssertionError: Got unexpecte...
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-8192128f] - AssertionEr...
[00:01:30] FAILED tests/test_speed.py::test_kem[Kyber512] - AssertionError: Got unexpect...
[00:01:30] FAILED tests/test_speed.py::test_kem[HQC-256] - AssertionError: Got unexpecte...
[00:01:30] FAILED tests/test_speed.py::test_kem[Kyber768] - AssertionError: Got unexpect...
[00:01:30] FAILED tests/test_speed.py::test_kem[sntrup761] - AssertionError: Got unexpec...
[00:01:30] FAILED tests/test_speed.py::test_kem[FrodoKEM-640-AES] - AssertionError: Got ...
[00:01:30] FAILED tests/test_speed.py::test_kem[FrodoKEM-640-SHAKE] - AssertionError: Go...
[00:01:30] FAILED tests/test_speed.py::test_kem[FrodoKEM-976-AES] - AssertionError: Got ...
[00:01:30] FAILED tests/test_speed.py::test_kem[FrodoKEM-1344-AES] - AssertionError: Got...
[00:01:30] FAILED tests/test_speed.py::test_sig[Dilithium3] - AssertionError: Got unexpe...
[00:01:30] FAILED tests/test_speed.py::test_kem[FrodoKEM-1344-SHAKE] - AssertionError: G...
[00:01:30] FAILED tests/test_speed.py::test_kem[FrodoKEM-976-SHAKE] - AssertionError: Go...
[00:01:30] FAILED tests/test_speed.py::test_sig[Dilithium5] - AssertionError: Got unexpe...
[00:01:30] FAILED tests/test_speed.py::test_sig[Falcon-512] - AssertionError: Got unexpe...
[00:01:30] FAILED tests/test_speed.py::test_sig[Dilithium2] - AssertionError: Got unexpe...
[00:01:30] FAILED tests/test_speed.py::test_sig[Falcon-1024] - AssertionError: Got unexp...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHA2-128s-simple] - AssertionEr...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHA2-128f-simple] - AssertionEr...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHA2-192f-simple] - AssertionEr...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHA2-192s-simple] - AssertionEr...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHA2-256f-simple] - AssertionEr...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHA2-256s-simple] - AssertionEr...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHAKE-128f-simple] - AssertionE...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHAKE-128s-simple] - AssertionE...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHAKE-192f-simple] - AssertionE...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHAKE-256f-simple] - AssertionE...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHAKE-256s-simple] - AssertionE...
[00:01:30] FAILED tests/test_speed.py::test_kem[Classic-McEliece-6688128] - AssertionErr...
[00:01:30] FAILED tests/test_speed.py::test_kem[Kyber1024] - AssertionError: Got unexpec...
[00:01:30] FAILED tests/test_speed.py::test_sig[SPHINCS+-SHAKE-192s-simple] - AssertionE...
[00:01:30] ================= 40 failed, 218 passed, 144 skipped in 40.35s =================
```
**Environment (please complete the following information):**
 - OS: `ALT Linux`
 - OpenSSL version `1.1.1t`
 - Compiler version used `gcc 12.2.1`
 - Build variables used `-DBUILD_SHARED_LIBS=ON -DOQS_DIST_BUILD=ON -DOQS_OPT_TARGET=generic`
 - liboqs version `0.8.0.rc1`

Full build log is temporary available there https://git.altlinux.org/tasks/321944/build/100/armh/log

I added `-DOQS_USE_RASPBERRY_PI` to `CFLAGS`[1] and after this all tests are passed.
```
====================== 258 passed, 144 skipped in 37.18s =======================
```

Thanks!

[1] Adding to `cmake` command does not work:
```
CMake Error: Parse error in command line argument: OQS_USE_RASPBERRY_PI
```

--- a/tests/ds_benchmark.h
+++ b/tests/ds_benchmark.h
@@ -128,13 +128,6 @@ static uint64_t _bench_rdtsc(void) {
 	__asm__ volatile(".byte 0x0f, 0x31"
 	                 : "=A"(x));
 	return x;
-#elif defined(__arm__) && !defined(_OQS_RASPBERRY_PI)
-	/* Use the ARM performance counters. */
-	unsigned int value;
-	/* Read CCNT Register */
-	__asm__ volatile("mrc p15, 0, %0, c9, c13, 0\t\n"
-	                 : "=r"(value));
-	return value;
 #elif defined(SPEED_USE_ARM_PMU)
 	/* Use the Performance Monitoring Unit */
 	uint64_t value;
@@ -154,33 +147,7 @@ static uint64_t _bench_rdtsc(void) {
 #endif
 }
 
-#if defined(__arm__) && !defined(_OQS_RASPBERRY_PI)
-static void _bench_init_perfcounters(int32_t do_reset, int32_t enable_divider) {
-	/* In general enable all counters (including cycle counter) */
-	int32_t value = 1;
-
-	/* Peform reset */
-	if (do_reset) {
-		value |= 2; /* reset all counters to zero */
-		value |= 4; /* reset cycle counter to zero */
-	}
-
-	if (enable_divider) {
-		value |= 8;    /* enable "by 64" divider for CCNT */
-	}
-
-	value |= 16;
-
-	/* Program the performance-counter control-register */
-	__asm__ volatile("mcr p15, 0, %0, c9, c12, 0\t\n" ::"r"(value));
-
-	/* Enable all counters */
-	__asm__ volatile("mcr p15, 0, %0, c9, c12, 1\t\n" ::"r"(0x8000000f));
-
-	/* Clear overflows */
-	__asm__ volatile("mcr p15, 0, %0, c9, c12, 3\t\n" ::"r"(0x8000000f));
-}
-#elif defined(SPEED_USE_ARM_PMU)
+#if defined(SPEED_USE_ARM_PMU)
 
 /* Enabling access to ARMv8's Performance Monitoring Unit
  * cannot be done from user mode. A kernel module to
@@ -205,16 +172,7 @@ static void _bench_init_perfcounters(void) {
     double _bench_cycles_x, _bench_cycles_mean, _bench_cycles_delta, _bench_cycles_M2, _bench_cycles_stdev; \
     double _bench_time_x, _bench_time_mean, _bench_time_delta, _bench_time_M2, _bench_time_stdev;
 
-#if defined(__arm__) && !defined(_OQS_RASPBERRY_PI)
-#define INITIALIZE_TIMER            \
-    _bench_init_perfcounters(1, 0); \
-    _bench_iterations = 0;          \
-    _bench_cycles_mean = 0.0;       \
-    _bench_cycles_M2 = 0.0;         \
-    _bench_time_cumulative = 0;     \
-    _bench_time_mean = 0.0;         \
-    _bench_time_M2 = 0.0;
-#elif defined(SPEED_USE_ARM_PMU)
+#if defined(SPEED_USE_ARM_PMU)
 #define INITIALIZE_TIMER        \
     _bench_init_perfcounters(); \
     _bench_iterations = 0;      \

